### PR TITLE
feat: disable 2fa

### PR DESCRIPTION
**Pull request by olawolejethro**
## What does the PR do?

This PR is about disabling two factor authentication for logged in users. The endpoint returns errors if:

- [x] User is not logged in.
- [x] User does not exist.
- [x] Validation errors on the user input is done

How should this be manually tested?

1. Login to a user account, and get the access token

2. Make a POST request to the `api/v1/auth/2fa/disable` endpoint with the following payload:

```json
Authorization: Bearer <access_token>

{
    "password": "jetpack123",
    "totp_code": "123456"
}
```

Successful Response

On a successful testimonial creation, the API should return a 201 CREATED status code.

```json
{
    "status_code": 200,
    "message": "2FA is already disabled",
    "data": {
        "is_2fa_enabled": false
    }
}
```

Authentication Error:

When the current user is not properly authenticated to change the organization detail:

```json
{
  "status": "error",
  "message": "User is currently unauthorized, kindly authenticate to continue",
  "status_code": 401
}
```

Invalid UserID Error

```
![wrong user ID image](https://github.com/user-attachments/assets/6f9b6f4b-ff0d-4fa7-828f-aa61feff8440)

Validation:

If any required field is missing or invalid, the response should return a 400 Bad Request status code with appropriate validation error messages.

```json
{
  "status": "error",
  "message": ["name must be a string", "content must be a string"],
  "status_code": "400"
}
```

Task

- [x] Create a POST endpoint UPDATE /api/v1/2FA/disable to disable two factor authentication for logged in users.
- [x] Implement server-side logic to handle successful and failed update requests.
- [x] Ensure the user has the proper authentication to create testimonials.
- [x] Update the user two factor authenticator details in the database.
- [x] Update API documentation to reflect the new endpoint and its usage.

Reference Issue: [Link to Issue](https://github.com/hngprojects/hng_boilerplate_nestjs/issues/266)## What does the PR do?

This PR is about authentication  for logged in users. The endpoint returns errors if:

- [x] User is not logged in.
- [x] User does not exist.
- [x] Validation errors on the user input is done

How should this be manually tested?

1. Login to a user account, and get the access token

2. Make a POST request to the `api/v1/auth/2fa/disable` endpoint with the following payload:

```json
Authorization: Bearer <access_token>
{
    "password": "jetpack123",
    "totp_code": "123456"
}
```

Successful Response

On a successful testimonial creation, the API should return a 201 CREATED status code.

```json
{
    "status_code": 200,
    "message": "2FA is already disabled",
    "data": {
        "is_2fa_enabled": false
    }
}
```

### CODE REVIEW

**Comment by TheCodeGhinux** [comment]
fix conflicts

File: db/migrations/1721901095345-migration.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class Migration1721901095345 implements MigrationInterface {
+  public async up(queryRunner: QueryRunner): Promise<void> {}
+
+  public async down(queryRunner: QueryRunner): Promise<void> {}
+}
```
Comment (davidolanrewaju) [inline db/migrations/1721901095345-migration.ts:7]: Remove the migration file, it's not needed

File: src/modules/auth/auth.service.ts [hunk 2]
```diff
@@ -224,4 +225,48 @@ export default class AuthenticationService {
...
+
+  public async disable2FA(user_id: string, password: string, totp_code: string) {
+    const { user, isValid, ...validationResponse } = await this.validateUserAndPassword(user_id, password);
+
+    if (!isValid) {
+      throw validationResponse;
+    }
+
+    // totp_code
+
```
Comment (davidolanrewaju) [inline src/modules/auth/auth.service.ts:237]: Remove uneccessary comments

File: src/modules/auth/dto/disable -2fa.dto.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import { IsString, IsNotEmpty, Length } from 'class-validator';
+
+export class Disable2FADto {
+  @ApiProperty()
+  @IsString()
+  password: string;
+
+  // @IsNotEmpty()
+  @ApiProperty()
+  @IsString()
```
Comment (davidolanrewaju) [inline src/modules/auth/dto/disable -2fa.dto.ts:11]: Remove uneccessary comments

File: src/modules/auth/dto/disable -2fa.dto.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  // @Length(6, 6)
+  totp_code: string;
+}
```
Comment (davidolanrewaju) [inline src/modules/auth/dto/disable -2fa.dto.ts:14]: Remove comments

**Pull request closed without merging**